source/common/http/conn_manager_impl.cc

```diff
@@ -679,7 +680,16 @@ void ConnectionManagerImpl::ActiveStream::onRequestTimeout() {
 void ConnectionManagerImpl::ActiveStream::onStreamMaxDurationReached() {
   ENVOY_STREAM_LOG(debug, "Stream max duration time reached", *this);
   connection_manager_.stats_.named_.downstream_rq_max_duration_reached_.inc();
-  connection_manager_.doEndStream(*this);
+  stream_info_.setResponseFlag(StreamInfo::ResponseFlag::DownstreamMaxStreamDurationReached);
+
+  // If downstream response had started, envoy don't return 408.
```

Comment:
This is a behavior change  / bug fix and not just a response flag change. Please add a release note and cleanup the PR description to fully follow the template.

Reply:
add release note.

Reply:
Heh, just discovered this myself and have a PR which also fixes, if you want to just handle the response flag here.
If you want to fix the local reply, you'll need to flag protect (As Matt said) and also please only use sendLocalReply - it handles the "reset or reply" logic internally now.

Reply:
Does it mean that we don't have to aware the downstream response_header status and just to call sendLocalReply?

Reply:
Correct, I think you can now revert this part of the change but I would confirm with @alyssawilk

Reply:
So you have two choices.  You can either revert this part of the change (set response flags, and let my WIP PR take care of the sendLocalResponse changes) or you can runtime-guard always calling sendLocalReply as I do there.  calling doEndStream is basically deprecated.

Reply:
https://github.com/envoyproxy/envoy/pull/12228 sent out before I got all the test cleaned up, but it has the sendLocalReply with guards I mentioned.